### PR TITLE
fix(CDU): suppress TMPY FPLN when no modification made in airways page

### LINKED ISSUE #8910: CDU: TMPY fpln appears after leaving airways page despite no modification being made
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-09-13T10:04:03+00:00",
    "ref": "refs/heads/master",
    "sha": "c4ab9895fb290e60a6c0bc80e809421b47f76bb7",
    "actor": "frankkopp",
    "event_name": "manual",
    "pretty_release_name": "master:c4ab9895",
    "version": "a32nx-v0.12.0-dev.c4ab989"
}
```

### Describe the bug

Opening airways pages and returning to FPLN without performing a modification causes TMPY FPLN to appear.
 

https://github.com/user-attachments/assets/c643abfd-bad2-46ad-813f-55c5199efd23

### Expected behavior

No TMPY FPLN to appear

### Steps to reproduce

See video

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

bruno_pt99

### PR DESCRIPTION
**Pull request by robertxing2004**
Fixes #8910 

## Summary of Changes

Corrected FMS logic to prevent TMPY FPLN from appearing after leaving airways page without making any modifications.

## Screenshots (if necessary)

Refer to #8910 for previous incorrect behaviour

Corrected:
https://github.com/user-attachments/assets/76c8f6e1-591a-43c0-9f33-521591564ba4

## Additional context

Discord username (if different from GitHub):
robeet

## Testing instructions

1. Load anywhere, and start a flight plan from the MCDU
2. Open lateral revision page for a waypoint -> open airways page
3. Without making any modifications, press LSK6L (RETURN)
4. Should return to lat rev page -> press LSK6L (RETURN) again
5. Should return to flight plan page -> verify no TMPY FPLN state

Secondary:
1. Following step 3 above, return to airways page (LSK5R) and make a modification, then insert
2. Observe TMPY FPLN and confirm insert
3. Verify new modifications inserted correctly

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

**Comment by tracernz** [comment]
How feasible is it to not actually create a TMPY until the first modification is made on the airways page. This feels like a workaround rather than actually fixing the root cause.

**Comment by robertxing2004** [comment]
As far as I can tell, it would at most require a rewrite of AirwaysFromWaypointPage. The current implementation starts a TMPY as soon as the page is opened, so it would just be moving that logic to only trigger once an airway (and terminating waypoint?) is entered. I can definitely take a stab at it.

**Comment by 2hwk** [comment]
Approved as a stopgap measure

**Review by 2hwk — APPROVED** [review]

**Review by tracernz — CHANGES_REQUESTED** [review]
Still needs changelog at least.

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -124,6 +124,7 @@
 1. [FMS] Run vertical predictions without V-speeds - @BlueberryKing (BlueberryKing)
 1. [ELEC] Use ADIRU and LGCIU signals for speed and in flight determination - @Gurgel100 (Pascal)
 1. [FMS] Do not transmit bearing information to ND on manual legs - @BravoMike99 (bruno_pt99)
+1. [MCDU] Suppress TMPY FPLN when no changes made on airways page - @robertxing2004 (robeet)
```
Comment (tracernz) [inline .github/CHANGELOG.md:127]: Wrong version, needs to be under 0.13.0 (you might need to rebase/merge master first).

Comment (robertxing2004) [reply .github/CHANGELOG.md:127]: Ah yes, you're right.

**Comment by robertxing2004** [comment]
A true fix (without changing anything in the fmgc code) requires validating the first airway entry without using a tmpy but I believe it's doable.

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -136,6 +137,7 @@
...
```
Comment (tracernz) [inline .github/CHANGELOG.md:140]: Still have to remove this one, then good to go.
```suggestion
```

**Review by 2hwk — APPROVED** [review]

**Comment by 2hwk** [comment]
Quality Assurance Tester Report

Discord Username: 2cas
Object of testing: #9148
Aircraft:  A32NX
Tier of Testing: 1
Date: 17/11/2024 

Testing Process: 
- [x] Load anywhere, and start a flight plan from the MCDU
- [x] Open lateral revision page for a waypoint -> open airways page
- [x] Without making any modifications, press LSK6L (RETURN)
- [x] Should return to lat rev page -> press LSK6L (RETURN) again
- [x] Should return to flight plan page -> verify no TMPY FPLN state

Secondary:

 - [x] Following step 3 above, return to airways page (LSK5R) and make a modification, then insert
 - [x] Observe TMPY FPLN and confirm insert
 - [x] Verify new modifications inserted correctly


Testing Results: 
QA Passed

Negatives:       
- As noted in the PR comments above, there is no further improvement other than what this PR specifically addresses, but also this is not a regression from master.

Conclusions: 
- LGTM

**Pull request merged**